indico/modules/rb_new/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -0,0 +1,659 @@
+import fetchRoomAvailabilityURL from 'indico-url:rooms_new.admin_room_availability';
+import fetchRoomEquipmentURL from 'indico-url:rooms_new.admin_room_equipment';
+import updateRoomBasicDetailsURL from 'indico-url:rooms_new.admin_update_room';
+import updateRoomEquipmentURL from 'indico-url:rooms_new.admin_update_room_equipment';
+import updateRoomAttributesURL from 'indico-url:rooms_new.admin_update_room_attributes';
+import updateRoomAvailabilityURL from 'indico-url:rooms_new.admin_update_room_availability';
+import _ from 'lodash';
+import React from 'react';
+import {bindActionCreators} from 'redux';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import {Button, Checkbox, Dimmer, Dropdown, Form, Grid, Header, Input, Loader, Message, Modal, TextArea} from 'semantic-ui-react';
+import {Form as FinalForm, Field} from 'react-final-form';
+import {FieldArray} from 'react-final-form-arrays';
+import arrayMutators from 'final-form-arrays';
+import shortid from 'shortid';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import {snakifyKeys, camelizeKeys} from 'indico/utils/case';
+import {getChangedValues, ReduxCheckboxField, ReduxFormField} from 'indico/react/forms';
+import {Translate} from 'indico/react/i18n';
+import PrincipalSearchField from 'indico/react/components/PrincipalSearchField';
+import EquipmentList from './EquipmentList';
+import DailyAvailability from './DailyAvailability';
+import NonBookablePeriods from './NonBookablePeriods';
+import SpriteImage from '../../components/SpriteImage';
+import {actions as roomsActions} from '../../common/rooms';
+import * as roomsSelectors from './selectors';
+import {selectors as userSelectors} from '../user';
+
+import './RoomEditModal.module.scss';
+
+
+function isInvalidNotificationPeriod(days) {
+    return days !== null && (days > 30 || days < 1);
+}
+
+function validate(fields) {
+    const {building, floor, number, capacity, attributes, surfaceArea, nonbookablePeriods, notificationBeforeDays,
+           notificationBeforeDaysWeekly, notificationBeforeDaysMonthly} = fields;
+    const errors = {};
+    if (!building) {
+        errors.building = Translate.string('Please provide a building.');
+    }
+    if (!floor) {
+        errors.floor = Translate.string('Please provide a floor.');
+    }
+    if (!number) {
+        errors.number = Translate.string('Please provide a number.');
+    }
+    if (!capacity) {
+        errors.capacity = Translate.string('Please provide capacity.');
+    }
+    if (capacity !== null && capacity < 1) {
+        errors.capacity = Translate.string('Please provide a valid capacity number.');
+    }
+    if (surfaceArea !== null && surfaceArea < 1) {
+        errors.surfaceArea = Translate.string('Please provide a valid surface area number.');
+    }
+    if (isInvalidNotificationPeriod(notificationBeforeDays)) {
+        errors.notificationBeforeDays = Translate.string('Number of days must be between [1, 30]');
```

Comment:
Perhaps me can rephrase it to "... between 1 and 30." IMHO it sounds better, we don't have to necessarily keep what we have in the old rb.

Reply:
Sounds better to me as well